### ISSUE TITLE
changing to --bundle-node-modules does not properly rebuild

### ISSUE THREAD

**Issue by asottile** [parcel-bundler/parcel#3841]
# 🐛 bug report

When invoking `parcel build` first without `--bundle-node-modules` and then again with `--bundle-node-modules` -- the node modules are not bundled due to the compilation being completely cached

## 🎛 Configuration (.babelrc, package.json, cli command)

```bash
npm install left-pad
npm install parcel-bundler
```

```js
{
  "private": true,
  "dependencies": {
    "left-pad": "^1.3.0"
  },
  "devDependencies": {
    "parcel-bundler": "^1.12.4"
  }
}
```

## 🤔 Expected Behavior

I expect to not need to delete the cache to change the output

## 😯 Current Behavior

```js
const leftPad = require('left-pad')

console.log(leftPad('ohai'));
```

```console
$ rm -rf dist .cache && ./node_modules/.bin/parcel build t.js -t node && cat dist/t.js && echo
✨  Built in 248ms.

dist/t.js        1.15 KB    97ms
dist/t.js.map      262 B     3ms
parcelRequire=function(e,r,t,n){var i,o="function"==typeof parcelRequire&&parcelRequire,u="function"==typeof require&&require;function f(t,n){if(!r[t]){if(!e[t]){var i="function"==typeof parcelRequire&&parcelRequire;if(!n&&i)return i(t,!0);if(o)return o(t,!0);if(u&&"string"==typeof t)return u(t);var c=new Error("Cannot find module '"+t+"'");throw c.code="MODULE_NOT_FOUND",c}p.resolve=function(r){return e[t][1][r]||r},p.cache={};var l=r[t]=new f.Module(t);e[t][0].call(l.exports,p,l,l.exports,this)}return r[t].exports;function p(e){return f(p.resolve(e))}}f.isParcelRequire=!0,f.Module=function(e){this.id=e,this.bundle=f,this.exports={}},f.modules=e,f.cache=r,f.parent=o,f.register=function(r,t){e[r]=[function(e,r){r.exports=t},{}]};for(var c=0;c<t.length;c++)try{f(t[c])}catch(e){i||(i=e)}if(t.length){var l=f(t[t.length-1]);"object"==typeof exports&&"undefined"!=typeof module?module.exports=l:"function"==typeof define&&define.amd?define(function(){return l}):n&&(this[n]=l)}if(parcelRequire=f,i)throw i;return f}({"aRuQ":[function(require,module,exports) {
const o=require("left-pad");console.log(o("ohai"));
},{}]},{},["aRuQ"], null)
//# sourceMappingURL=/t.js.map
$ ./node_modules/.bin/parcel build t.js -t node --bundle-node-modules && cat dist/t.js && echo
✨  Built in 145ms.

dist/t.js        1.15 KB    4ms
dist/t.js.map      262 B    5ms
parcelRequire=function(e,r,t,n){var i,o="function"==typeof parcelRequire&&parcelRequire,u="function"==typeof require&&require;function f(t,n){if(!r[t]){if(!e[t]){var i="function"==typeof parcelRequire&&parcelRequire;if(!n&&i)return i(t,!0);if(o)return o(t,!0);if(u&&"string"==typeof t)return u(t);var c=new Error("Cannot find module '"+t+"'");throw c.code="MODULE_NOT_FOUND",c}p.resolve=function(r){return e[t][1][r]||r},p.cache={};var l=r[t]=new f.Module(t);e[t][0].call(l.exports,p,l,l.exports,this)}return r[t].exports;function p(e){return f(p.resolve(e))}}f.isParcelRequire=!0,f.Module=function(e){this.id=e,this.bundle=f,this.exports={}},f.modules=e,f.cache=r,f.parent=o,f.register=function(r,t){e[r]=[function(e,r){r.exports=t},{}]};for(var c=0;c<t.length;c++)try{f(t[c])}catch(e){i||(i=e)}if(t.length){var l=f(t[t.length-1]);"object"==typeof exports&&"undefined"!=typeof module?module.exports=l:"function"==typeof define&&define.amd?define(function(){return l}):n&&(this[n]=l)}if(parcelRequire=f,i)throw i;return f}({"aRuQ":[function(require,module,exports) {
const o=require("left-pad");console.log(o("ohai"));
},{}]},{},["aRuQ"], null)
//# sourceMappingURL=/t.js.map
$ rm -rf .cache && ./node_modules/.bin/parcel build t.js -t node --bundle-node-modules && cat dist/t.js && echo
✨  Built in 330ms.

dist/t.js.map    2.08 KB      9ms
dist/t.js        1.48 KB    172ms
parcelRequire=function(e,r,t,n){var i,o="function"==typeof parcelRequire&&parcelRequire,u="function"==typeof require&&require;function f(t,n){if(!r[t]){if(!e[t]){var i="function"==typeof parcelRequire&&parcelRequire;if(!n&&i)return i(t,!0);if(o)return o(t,!0);if(u&&"string"==typeof t)return u(t);var c=new Error("Cannot find module '"+t+"'");throw c.code="MODULE_NOT_FOUND",c}p.resolve=function(r){return e[t][1][r]||r},p.cache={};var l=r[t]=new f.Module(t);e[t][0].call(l.exports,p,l,l.exports,this)}return r[t].exports;function p(e){return f(p.resolve(e))}}f.isParcelRequire=!0,f.Module=function(e){this.id=e,this.bundle=f,this.exports={}},f.modules=e,f.cache=r,f.parent=o,f.register=function(r,t){e[r]=[function(e,r){r.exports=t},{}]};for(var c=0;c<t.length;c++)try{f(t[c])}catch(e){i||(i=e)}if(t.length){var l=f(t[t.length-1]);"object"==typeof exports&&"undefined"!=typeof module?module.exports=l:"function"==typeof define&&define.amd?define(function(){return l}):n&&(this[n]=l)}if(parcelRequire=f,i)throw i;return f}({"msLR":[function(require,module,exports) {
"use strict";module.exports=t;var r=[""," ","  ","   ","    ","     ","      ","       ","        ","         "];function t(t,e,n){if((e-=(t+="").length)<=0)return t;if(n||0===n||(n=" ")," "===(n+="")&&e<10)return r[e]+t;for(var u="";1&e&&(u+=n),e>>=1;)n+=n;return u+t}
},{}],"aRuQ":[function(require,module,exports) {
const o=require("left-pad");console.log(o("ohai"));
},{"left-pad":"msLR"}]},{},["aRuQ"], null)
//# sourceMappingURL=/t.js.map
```

## 💁 Possible Solution

🤷‍♂ 

## 🔦 Context

I was trying to use this tool for the first time to bundle some stuff for github actions and was trying to learn what the various command line options do (and was very surprised when they seemed to do nothing).

## 💻 Code Sample

See above

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.12.4
| Node             | 12.13.0
| npm/Yarn         | 6.12.0
| Operating System | ubuntu 18.04

**Comment by asottile** [comment]
this is still an issue

**Comment by tscpp** [comment]
Please add an option to bundle every thing

**Comment by devongovett** [comment]
In parcel 2 bundling node modules is the default unless you're building a library. You can also configure this: https://v2.parceljs.org/configuration/package-json/#includenodemodules.

**Comment by asottile** [comment]
unless I'm missing something this seems to still be a problem:

```console
$ cat package.json 
{
  "private": true,
  "dependencies": {
    "left-pad": "^1.3.0"
  },
  "devDependencies": {
    "parcel": "^2.0.0-beta.1"
  },
  "targets": {
    "t": {
      "engines": {"node": ">=4.x"},
      "includeNodeModules": false
    }
  }
}
$ ./node_modules/.bin/parcel build t.js && cat dist/t.js && echo
✨ Built in 1.62s

dist/t.js    80 B    352ms
var e=require("left-pad");console.log(e("ohai"));
//# sourceMappingURL=t.js.map

$ babi package.json   # editing out the includeNodeModules setting

$ cat package.json 
{
  "private": true,
  "dependencies": {
    "left-pad": "^1.3.0"
  },
  "devDependencies": {
    "parcel": "^2.0.0-beta.1"
  },
  "targets": {
    "t": {
      "engines": {"node": ">=4.x"}
    }
  }
}
$ ./node_modules/.bin/parcel build t.js && cat dist/t.js && echo
✨ Built in 1.39s

dist/t.js    80 B    273ms
var e=require("left-pad");console.log(e("ohai"));
//# sourceMappingURL=t.js.map

```

**Comment by devongovett** [comment]
Ah, sorry. It will be fixed after #5212 is merged. I'm living in the future. 😄

**Comment by asottile** [comment]
still broken on 2.0.0-beta2 (which if I'm reading github properly it is after #5212 was merged)

**Comment by mischnic** [comment]
@asottile Can you please open a new issue with that code example?

**Comment by mischnic** [comment]
The reason for this behaviour is that a node engine implies `"includeNodeModules": false`
So
```json
  "targets": {
    "t": {
      "engines": {
        "node": ">=4.x"
      },
      "includeNodeModules": false
    }
  }
```
and
```json
  "targets": {
    "t": {
      "engines": {
        "node": ">=4.x"
      }
    }
  }
```
do the same thing (even with `--no-cache`).

Changing between `"includeNodeModules": false` and `"includeNodeModules": true` works correctly for me.

**Comment by asottile** [comment]
I see, that works
